### DISCUSSION TITLE
Video MIME type error despite accepted video codec - audio issue?

### DISCUSSION THREAD

**Discussion by rjcommanddfo** [biigle/core#985] [Q&A]
I received the MIME Type error after uploading a video file that has a valid web-playable codec (mp4 H.264). I used VLC to "convert" the codec, uploaded the video file, and it worked.

As far as I can tell, the only thing that was changed was the order of the streams and the _audio_ codec, which appears to have solved the problem. I'm wondering if I'm missing something, or if other folks have had similar issues with audio codecs & BIIGLE? Audio is not important in these videos and I'm happy to remove it from the files, but just curious!
![audio-codec-issue](https://github.com/user-attachments/assets/afd950b7-2535-4620-b722-05a2933ffeb8)

**Comment by mzur** [comment]
What is the output of:
```
ffprobe -v error -select_streams v:0 -show_entries stream=codec_name -of default=noprint_wrappers=1:nokey=1 'VIDEO_FILE'
```
for both files? Is both `h264`?

**Reply by rjcommanddfo** [reply]
Hi Martin, sorry for the delay - yes, both return `h264`!

**Reply by mzur** [reply]
@rjcommanddfo Below we figured out that the incorrect MIME type came from S3. How do you store your files? Do you see the MIME type error in the create volume form or in the video annotation tool?

**Reply by rjcommanddfo** [reply]
Interesting! The files are stored in MS Azure. No issues in the create volume form, I only see the MIME type error in the video annotation tool.

**Reply by mzur** [reply]
Then this is a different issue than reported below. Maybe you could also send me (links to) the two videos to info@biigle.de?

**Comment by barefooted23** [comment]
I've been having issues uploading mp4 files as well, even though they seem to have the correct codec. I also used VLC and eventually found a combo of codec and wrapper that worked, but it's been frustrating since the error requests an H264 codec when that's what the codec is in the files.

**Reply by mzur** [reply]
Could you please send an example file as a download link to info@biigle.de? Once we can reproduce the issue with a file, we can fix it.

**Reply by barefooted23** [reply]
email sent :)

**Reply by mzur** [reply]
For the record: This issue is caused by an incorrect MIME type of the file in S3. It can be fixed by specifying the correct MIME/content type during the S3 upload.
